codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/AwsEndpointBuiltins.java

```diff
@@ -32,6 +39,42 @@ public class AwsEndpointBuiltins implements GoIntegration {
             goTemplate("$T(options.UseARNRegion)", SdkGoTypes.Aws.Bool);
     private static final GoWriter.Writable BindAwsS3DisableMultiRegionAccessPoints =
             goTemplate("$T(options.DisableMultiRegionAccessPoints)", SdkGoTypes.Aws.Bool);
+    private static final GoWriter.Writable BindAccountID =
+            goTemplate("resolveAccountID(getIdentity(ctx), options.AccountIDEndpointMode)");
+
+    @Override
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoDelegator goDelegator
+    ) {
+        if (!settings.getService(model).hasTrait(EndpointRuleSetTrait.class)) {
+            return;
+        }
+        goDelegator.useShapeWriter(settings.getService(model), goTemplate("""
+        func resolveAccountID(identity $auth:T, mode $accountIDEndpointMode:T) *string {
+            if mode == $aidModeDisabled:T || mode == $aidModeUnset:T {
```

Comment:
An unset value is supposed to imply preferred, according to the spec:

> preferred, or no value - Default behavior

So this should only check for disabled